trio/_core/_multierror.py

```diff
@@ -322,8 +381,8 @@ class NonBaseMultiError(MultiError, ExceptionGroup):
 
 if have_tproxy:
     # http://doc.pypy.org/en/latest/objspace-proxies.html
-    def copy_tb(base_tb, tb_next):
-        def controller(operation):
+    def copy_tb(base_tb: TracebackType, tb_next: TracebackType | None) -> tputil:
```

Comment:
The return type here should still be `TracebackType` (casting the "proxy"). [`make_proxy()`](https://doc.pypy.org/en/latest/objspace-proxies.html#make_proxy) produces an object that seems to be the object passed in, but can use the controller callable to override behaviours. So typing-wise it is a traceback.

Reply:
Changed in [e0f7a9c](https://github.com/python-trio/trio/pull/2742/commits/e0f7a9cac32647c303af28f101600ade0331a45e)